web/src/components/apps/EmbeddedClusterManagement.tsx

```diff
@@ -233,6 +233,16 @@ const EmbeddedClusterManagement = ({
   // #region node type logic
   const NODE_TYPES = ["controller"];
 
+  useEffect(() => {
+    const nodeTypes = rolesData?.roles || NODE_TYPES;
+    if (nodeTypes.length === 1) {
+      // if there's only one node type, select it by default
+      setSelectedNodeTypes(nodeTypes);
+    } else {
+      setSelectedNodeTypes([]);
```

Comment:
Since `selectedNodeTypes` is already initialized to an empty array, is this needed?

Reply:
my thought was if the `rolesData` was refetched and contained more than one role, then we would want to deselect. though it doesn't look like we are refetching this data ever, so i do think we can remove that

Reply:
That's fair. If you think we aren't refetching this data ever, I'd say it's safe to remove, but I also understand that logic.

Reply:
i'll go ahead and remove it for now as it seems we just fetch it when the component loads

Reply:
it seems that removing this may actually be causing an issue since `rolesData` is initially `undefined`, so we fall back to `NODE_TYPES` in that case. since that only has one role, we select it by default. then once `rolesData` is fetched, we don't deselect and still have that initial role from `NODE_TYPES` selected. going to add that `else` back in